Src/Common/AppInsightsActivity.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+
+    //this is a temporary solution that mimics System.DiagnosticSource,Activity and Correlation HTTP protocol:
+    //https://github.com/lmolkova/correlation/blob/master/http_protocol_proposal_v1.md
+    //It does not implement 
+    // - Request-Id length limitation and overflow
+    internal class AppInsightsActivity
```

Comment:
It makes sense to have it in Base SDK for 4.0/4.5 if we also have Activity for 4.6 there. Let's decide on it first.